base/strings/basic.jl

```diff
@@ -780,3 +780,16 @@ julia> codeunits("Juλia")
 ```
 """
 codeunits(s::AbstractString) = CodeUnits(s)
+
+function _split_rest(s::AbstractString, n::Int)
+    lastind = lastindex(s)
+    i = try
+        prevind(s, lastind, n)
+    catch e
+        e isa BoundsError || rethrow()
+        @assert _check_length_split_rest(length(s), n)
+    end
+    last_n = SubString(s, nextind(s, i), lastind)
+    front = s[begin:i]
```

Comment:
Is there a reason not to use either indexing or SubString for both values?

Reply:
I think my reasoning was that the type of `last_n` doesn't really matter as long as it iterates correctly, so we might as well avoid the extra allocation. I guess we could make `front` a `SubString` as well, but that might be wasteful for small string if `n` is not much smaller than the string length, since we can't free the original string